CPT331.iOS/CPT331.iOS/LocationManager.swift

```diff
@@ -0,0 +1,170 @@
+//
+//  LocationManager.swift
+//  SpatioNews
+//
+//  Created by Peter Weller on 16/09/2016.
+//  Copyright © 2016 Peter Weller. All rights reserved.
+//
+
+import Foundation
+import CoreData
+import CoreLocation
+import Mapbox
+import MapboxGeocoder
+
+class LocationManager {
+    
+    internal static var lastSearchTask: NSURLSessionDataTask?
+    
+    //        let nsw = State.create("New South Wales", abbreviation: "NSW")
+    //        let qld = State.create("Queensland", abbreviation: "QLD")
+    //        let sa = State.create("South Australia", abbreviation: "SA")
+    //        let tas = State.create("Tasmania", abbreviation: "TAS")
+    //        let vic = State.create("Victoria", abbreviation: "VIC")
+    //        let wa = State.create("Western Australia", abbreviation: "WA")
+    //        let act = State.create("Australian Capital Territory", abbreviation: "ACT")
+    //        let nt = State.create("Northern Territory", abbreviation: "NT")
+    
+    
+    //        if let path = NSBundle.mainBundle().pathForResource("NSW_Places", ofType: "csv") {
+    //            let importer = CSVImporter<[String]>(path: path)
+    //            importer.startImportingRecords { $0 }.onFinish { importedRecords in
+    //                for record in importedRecords {
+    //                    if let lat = Double(record[3]), let lng = Double(record[4]) {
+    //                        Suburb.create(record[1], state: nsw!, latitude: lat, longitude: lng)
+    //                    }
+    //                }
+    //            }
+    //        }
+    
+    
+    class func getSuburbs(usingSearchQuery query:String, relativeToLocation location: CLLocation?) -> [CDSuburb] {
+        let fetchRequest = NSFetchRequest(entityName: "Suburb")
+        fetchRequest.predicate = NSPredicate(format: "(name CONTAINS[cd] %@)", query)
+        
+        do {
+            if let suburbs = try NSManagedObject.getMainContext().executeFetchRequest(fetchRequest) as? [CDSuburb] {
+
+                // If location data is available, sort locations relative to distance from user
+                if location != nil {
```

Comment:
If you like, the web API stuff can do all the sorting for you if you supply the user's location. Might be nicer to have the backend do the heavy lifting for you?

That being said, if you are offering several sorting options on the client, then you'll need to do this anyway.